### PR TITLE
Installer changes

### PR DESCRIPTION
**Pull request by luislopes-outsystems**

### CODE REVIEW

File: src/Outsystems.SetupTools/Lib/PlatformSetup.ps1 [hunk 1]
```diff
@@ -166,12 +239,12 @@ function InstallDotNetCore([string]$Sources)
 {
     if($Sources)
     {
-        $installer = "$Sources\DotNetCore_2_WindowsHosting.exe"
+        $installer = "$Sources\DotNetCore_WindowsHosting.exe"
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Lib/PlatformSetup.ps1]: We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

File: src/Outsystems.SetupTools/Lib/PlatformSetup.ps1 [hunk 2]
```diff
@@ -166,12 +239,12 @@ function InstallDotNetCore([string]$Sources)
...
         LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 2 -Message "Using local file: $installer"
     }
     else
     {
-        $installer = "$ENV:TEMP\DotNetCore_2_WindowsHosting.exe"
+        $installer = "$ENV:TEMP\DotNetCore_WindowsHosting.exe"
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Lib/PlatformSetup.ps1]: (same as above) We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

File: src/Outsystems.SetupTools/Lib/PlatformSetup.ps1 [hunk 3]
```diff
@@ -66,9 +86,16 @@ function GetDotNet4Version()
 {
     LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 2 -Message "Getting the registry value HKLM:SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\<langid>\Release."
 
+    <#
+        RPD-4212: For Windows installations with Japanese language, registry has two entries located at
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Lib/PlatformSetup.ps1]: Should we commit to the public repo the internal issue number? I don't actually see any problem with this, just making a note.

File: src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1 [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+                {
+                    $NETConfig = $(GetDotNetLimits)
+
+                    $CurrentMaxRequestLength = $NETConfig.SystemWeb.HttpRuntime.maxRequestLength
+                    $CurrentMaxRequestLengthInMB = "$($CurrentMaxRequestLength/1024) MB"
+                    $CurrentExecutionTimeout = $NETConfig.SystemWeb.HttpRuntime.executionTimeout
+
+                    # upload size limits default is 4096 KB
+                    #   https://docs.microsoft.com/en-us/dotnet/api/system.web.configuration.httpruntimesection.maxrequestlength
+                    $DefaultMaxRequestLength = 4096
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1]: We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

File: src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1 [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+                    $DefaultMaxRequestLengthInMB = "$($DefaultMaxRequestLength/1024) MB"
+
+                    # execution timeout default is 110 seconds
+                    #   https://docs.microsoft.com/en-us/dotnet/api/system.web.configuration.httpruntimesection.executiontimeout
+                    [TimeSpan]$DefaultExecutionTimeout = "00:01:50"
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1]: (same as above) We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

File: src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1 [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+ 
+        $PerfTuning.Sections.ProcessSchedulingConfig = CreatePerfTuningStatus -Title "Setting Windows processor scheduling priority to 'background services'" `
+            -ScriptBlock `
+            {
+                $ShouldBeSkipped = $True
+                $Messages = @()
+
+                try
+                {
+                    $Value = RegRead -Path "HKLM:\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\PriorityControl" -Name "Win32PrioritySeparation" -Type "Dword"
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1]: We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

File: src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1 [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+
+                    $ShouldBeSkipped = ($Value -eq 24)
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1]: (same as above) We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

File: src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1 [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+
+                try
+                {
+                    $Filter = "system.webServer/security/requestFiltering/requestLimits"
+                    $Name = "maxAllowedContentLength"
+
+                    $MaxAllowedContentLength = $(GetWebConfigurationProperty -PSPath "MACHINE/WEBROOT/APPHOST" -Filter "$Filter" -Name "$Name")
+                    # MaxAllowedContentLength default value is 30000000
+                    #  https://docs.microsoft.com/en-us/iis/configuration/system.webserver/security/requestfiltering/requestlimits/#configuration
+                    $DefaultMaxAllowedContentLength = 30000000
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1]: (same as above) We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

File: src/Outsystems.SetupTools/Functions/Set-OSServerPerformanceTunning.ps1 [hunk 9]
```diff
@@ -52,165 +138,215 @@ function Set-OSServerPerformanceTunning
...
-        catch
+        if (-not $SkipProcessSchedulingConfig)
         {
-            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Exception $_.Exception -Stream 3 -Message "Error configuring processor scheduling priority"
-            WriteNonTerminalError -Message "Error configuring processor scheduling priority"
+            # Configure process scheduling -- http://technet.microsoft.com/library/Cc976120
+            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message "Configuring Windows processor scheduling priority to background services"
+            try
+            {
+                RegWrite -Path "HKLM:\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\PriorityControl" -Name "Win32PrioritySeparation" -Type "Dword" -Value 24
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Functions/Set-OSServerPerformanceTunning.ps1]: We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

Note: This maps to the registry path/value in src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1

File: src/Outsystems.SetupTools/Functions/Set-OSServerPerformanceTunning.ps1 [hunk 10]
```diff
@@ -52,165 +138,215 @@ function Set-OSServerPerformanceTunning
...
+                $MaxAllowedContentLength = $OSPerfTuningMaxAllowedContentLength
+            }
 
-        # Configure IIS Application Pools
-        $DefaultWebSiteApps = $(Get-WebApplication -Site "Default Web Site").Path
+            # Configure IIS requests limits (Server Level)
+            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message "Configuring IIS upload size limits (maxAllowedContentLength = $MaxAllowedContentLength)"
+            try
+            {
+                SetWebConfigurationProperty -PSPath "MACHINE/WEBROOT/APPHOST" -Filter "system.webServer/security/requestFiltering/requestLimits" -Name "maxAllowedContentLength" -Value $MaxAllowedContentLength
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Functions/Set-OSServerPerformanceTunning.ps1]: We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

Note: This maps to the registry pathin src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1

File: src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1 [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+
+        $PerfTuning.Sections.IISConnectionsConfig = CreatePerfTuningStatus -Title "IIS for unlimited connections configuration" `
+            -ScriptBlock `
+            {
+                $ShouldBeSkipped = $True
+                $Messages = @()
+
+                try
+                {
+                    $IISConfigs = $(GetWebConfigurationProperty -PSPath "IIS:\" -Filter "system.applicationHost/sites/site[@name='Default Web Site']" -Name "Limits")
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1]: (not so sure about this one) We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

File: src/Outsystems.SetupTools/Functions/Set-OSServerPerformanceTunning.ps1 [hunk 12]
```diff
@@ -52,165 +138,215 @@ function Set-OSServerPerformanceTunning
...
-            # Build an array with all matching Apps.
-            foreach ($appMatchPattern in $($Config.Match))
+        if (-not $SkipIISConnectionsConfig)
+        {
+            # Configure IIS for unlimited connections. (Default Web Site) - This should not be needed cause IIS defaults to maximum.
+            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message "Configuring IIS for unlimited connections (MaxConnections = $OSPerfTuningMaxConnections)"
+            try
             {
-                $interestingApps += $DefaultWebSiteApps | Where-Object -FilterScript {$_ -like $appMatchPattern}
+                SetWebConfigurationProperty -PSPath "IIS:\" -Filter "system.applicationHost" -Name "sections['webLimits'].OverrideModeDefault" -Value "Allow"
```
Comment (fmalheiro) [inline src/Outsystems.SetupTools/Functions/Set-OSServerPerformanceTunning.ps1]: (again, not so sure about these) We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

Note: This maps to the registry path/value in src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1

**Comment by pintonunes** [comment]
Guys, I am going to split this PR in multiple ones. @fmalheiro, will probably need your help to make sure  we don't brake anything..

File: src/Outsystems.SetupTools/Functions/Get-OSServerPreReqs.ps1 [hunk 13]
```diff
@@ -0,0 +1,309 @@
+function Get-OSServerPreReqs
```
Comment (pintonunes) [inline src/Outsystems.SetupTools/Functions/Get-OSServerPreReqs.ps1]: @fmalheiro, @luislopes-outsystems Think we need to discuss this function. 
For me this should return an object with the pre-reqs installation status. IIS installed (true/false), .NET (true/false and so on)
Then the consumer ( Get-OutSystemsInstalledPrerequisites.ps1 ) should get that object and output whatever it needs to output to the installer window. 
This module should be agnostic and not tailored to a specific use case (ex: OutSystems installer).

**Comment by pintonunes** [comment]
Guys, as discussed with everyone I will close this PR and open a new one to have a fresh view of the changes.

**Pull request closed without merging**